src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,115 @@
+      mockSession = Mockito.mock(HttpSession.class);
+      Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+
+      mockResponse = Mockito.mock(HttpServletResponse.class);
+
+      mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+      Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/admin.jsp"))
+          .thenReturn(mockRequestDispatcher);
+
+      mockUserStore = Mockito.mock(UserStore.class);
+      AdminServlet.setUserStore(mockUserStore);
+
+      mockConversationStore = Mockito.mock(ConversationStore.class);
+      AdminServlet.setConversationStore(mockConversationStore);
+
+      mockMessageStore = Mockito.mock(MessageStore.class);
+      AdminServlet.setMessageStore(mockMessageStore);
+    }
+
+  @Test
+  public void testDoGet() throws IOException, ServletException {
+
+    List<User> fakeUserList = new ArrayList<>();
+    User fakeUser = new User(UUID.randomUUID(), "test_username", "fakePasswordHash", Instant.now());
+    fakeUser.setAdmin(true);
+    fakeUserList.add(fakeUser);
+
+    Mockito.when(mockUserStore.getAllUsers()).thenReturn(fakeUserList);
+
+    UUID fakeConversationId = UUID.randomUUID();
+    List<Conversation> fakeConversationList = new ArrayList<>();
+    fakeConversationList.add(
+        new Conversation(UUID.randomUUID(), UUID.randomUUID(), "test_conversation", Instant.now()));
+    Mockito.when(mockConversationStore.getAllConversations()).thenReturn(fakeConversationList);
+
+    List<Message> fakeMessageList = new ArrayList<>();
+    fakeMessageList.add(
+        new Message(UUID.randomUUID(),fakeConversationId, UUID.randomUUID(),"test_message",Instant.now()));
+    Mockito.when(mockMessageStore.getAllMessages()).thenReturn(fakeMessageList);
+    AdminServlet.doGet(mockRequest, mockResponse);
+
+    //Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+
+  }
+
+  @Test
+  public void testDoGet_UserNotLoggedIn() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn(null);
+
+    AdminServlet.doGet(mockRequest, mockResponse);
+    Mockito.verify(mockResponse).sendRedirect("/");
+  }
+
+  @Test
+  public void testDoGet_InvalidUser() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_username");
+    Mockito.when(mockUserStore.getUser("test_username")).thenReturn(null);
+
+    AdminServlet.doGet(mockRequest, mockResponse);
+    Mockito.verify(mockResponse).sendRedirect("/");
+  }
+
+  @Test
+  public void testDoGet_isAdmin() throws IOException, ServletException {
+  Mockito.when(mockRequest.getSession().getAttribute("user")).thenReturn("test_username");
+
+    int fakenumOfUsers = mockUserStore.getAllUsers().size();
+    int fakenumOfConvos = mockConversationStore.getAllConversations().size();
+    int fakenumOfMessages = mockMessageStore.getAllMessages().size();
+
+    AdminServlet.doGet(mockRequest, mockResponse);
```

Comment:
wait should you be verifying anything here? I also don't understand what this is really testing.